model/pmp/pmp_control.sail

```diff
@@ -97,7 +97,7 @@ function pmpCheck forall 'n, 0 < 'n <= max_mem_access . (
 
   let width : xlenbits = to_bits(width);
 
-  foreach (i from 0 to 63) {
+  foreach (i from 0 to sys_pmp_count - 1) {
```

Comment:
Kind of surprising that this compiles, e.g. when `sys_pmp_count` is 0. What does `foreach (i from 0 to -1)` do?

It doesn't actually matter for us since this is never called when `sys_pmp_count` is 0, but the Sail compiler doesn't know that.

Perhaps we want an `assert(sys_pmp_count > 0)` here just in case?

Reply:
> What does `foreach (i from 0 to -1)` do?

The body of the loop is not executed, as one would expect.

> It doesn't actually matter for us since this is never called when `sys_pmp_count` is 0, but the Sail compiler doesn't know that.
> 
> Perhaps we want an `assert(sys_pmp_count > 0)` here just in case?

I'm not convinced the assert is needed.  I'd argue it's actually simpler if we remove the `if sys_pmp_count == 0` check in `sys/mem.sail:phys_access_check()` and have this loop execute 0 times when `sys_pmp_count` is 0.